### ISSUE TITLE
Misleading setup

### ISSUE THREAD

**Issue by ifeherva** [littleredxh/DREML#6]
The paper clearly states that the authors used the same setup as in the proxy paper, yet the loss is not NCA but softmax:

https://github.com/littleredxh/DREML/blob/master/_code/Loss.py#L28

Though the difference is just one term it actually changes performance quite a bit.

**Comment by ifeherva** [comment]
The paper says

> We adopt a standard data augmentation scheme (random horizontal flip and random crops padded by 10 pixels on each side)

Yet, you add random rotations? Isn't that cheating?
https://github.com/littleredxh/DREML/blob/master/_code/Train.py#L76

**Comment by littleredxh** [comment]
> The paper clearly states that the authors used the same setup as in the proxy paper, yet the loss is not NCA but softmax:
> 
> https://github.com/littleredxh/DREML/blob/master/_code/Loss.py#L28
> 
> Though the difference is just one term it actually changes performance quite a bit.

It's the same operation as the original paper's equation 4

**Comment by littleredxh** [comment]
> The paper says
> 
> > We adopt a standard data augmentation scheme (random horizontal flip and random crops padded by 10 pixels on each side)
> 
> Yet, you add random rotations? Isn't that cheating?
> https://github.com/littleredxh/DREML/blob/master/_code/Train.py#L76

This is what I later add on. sorry for misleading.  You can remove it, but it doesn't improve so much.

**Comment by ifeherva** [comment]
> > The paper clearly states that the authors used the same setup as in the proxy paper, yet the loss is not NCA but softmax:
> > https://github.com/littleredxh/DREML/blob/master/_code/Loss.py#L28
> > Though the difference is just one term it actually changes performance quite a bit.
> 
> It's the same operation as the original paper's equation 4

Not exactly, Z in that equation is all negatives. In your case it is all (as in softmax CE)

**Comment by littleredxh** [comment]
dot-product and distance metric are reversed. If two images are similar, their dot-product is high and the distance is small. If two images are not similar, their dot-product is low and the distance is high. The proxy paper equation 4 use the distance metric, so all d should be negative.

**Comment by ifeherva** [comment]
I think you are missing the point. I am referring to the negative pairs (not the sign of d).

**Comment by littleredxh** [comment]
OK, you should be correct for the original paper's equation 4. But the origin NCA paper has the positive term in the denominator to make the output have the probabilistic meaning. Maybe it should be a mistake of the proxy paper. 

In the idea level, I followed the proxy paper. In the equation level, I followed the idea of original NCA paper.
